tests/unit/readers/test_ppf_reader.py

```diff
@@ -841,7 +827,8 @@ def test_get_signal_from_cache(
 def test_cache_read_bad_permissions():
     """Check that reading cached data fails if other users are allowed to
     write to the file. This is done for security reasons."""
-    reader = PPFReader(0, 0.0, 0.0)
+    with patch("indica.readers.ppfreader.SALClient", JETFakeSALClient):
+        reader = PPFReader(0, 0.0, 0.0)
```

Comment:
used `patched_ppf_reader`.

Reply:
Can't since `patched_ppf_reader` mocks `_write_cached_ppf `, whose behaviour is being tested.